statsmodels/sandbox/mice/mice.py

```diff
@@ -0,0 +1,1167 @@
+data set.
+
+4. Repeat steps 1-3 multiple times and combine the results using a
+combining rule.
+
+The specific way that each variable is imputed is specified using a
+conditional model and formula for each variable.  The default model is
+OLS, with a formula specifying main effects for all other variables.
+
+If the goal is only to produce imputed data sets, the MICE_data
+class can be used to wrap a data frame, providing facilities for doing
+the imputation.  Summary plots are available for assessing the
+performance of the imputation.
+
+If the imputed data sets are to be used to specify an additional
+'analysis model', a MICE instance can be used.  After specifying the
+MICE instance and running it, the results are combined using the
+`combine` method.  Results and various summary plots are then
+available.
+
+Terminology
+-----------
+The MICE procedure is determined by a family of conditional models.
+There is one conditional model for each variable with missing values.
+A conditional model may be conditioned on all or a subset of the
+remaining variables, using main effects, transformations,
+interactions, etc. as desired.
+
+A 'perturbation method' is a method for setting the parameter estimate
+in a conditional model.  The 'gaussian' perturbation method first fits
+the model (usually using maximum likelihood, but it could use any
+statsmodels fit procedure), then sets the parameter vector equal to a
+draw from the Gaussian approximation to the sampling distribution for
+the fit.  The 'bootstrap' perturbation method sets the parameter
+vector equal to a fitted parameter vector obtained when fitting the
+conditional model to a bootstrapped version of the data set.
+
+Class structure
+---------------
+There are three main classes in the module:
+
+* 'MICE_data' wraps a dataframe (or comparable data container),
+  incorporating information about the conditional models for each
+  variable with missing values. It can be used to produce multiply
+  imputed data sets that are to be further processed or distributed to
+  other researchers.  A number of plotting procedures are provided to
+  visualize the imputations.  The `history_func` hook allows any
+  features of interest of the imputed data sets to be saved for
+  further analysis.
+
+* 'MICE_model' takes a MICE_data object along with a specification for
+  an additional 'analysis model' which is the central model of
+  interest (it may or may not be identical to the conditional model
+  for the same outcome variable).  MICE_model carries out multiple
+  imputation using the conditional models, and fits the analysis model
+  to a subset of these imputed data sets.  It returns the fitted model
+  results for these analysis models.  It is structured as an iterator,
+  so the analysis model results are obtained using `next` or via any
+  standard python iterator pattern.
+
+* 'MICE' takes both a 'MICE_data' object and an analysis model
```

Comment:
something like `MICE` is the wrapper class that does the full analysis, ....
IIUC, to get the contrast between the function that holds some pieces and the top class